packages/transformer/src/test/standalone/IModelTransformer.test.ts

```diff
@@ -2655,6 +2655,262 @@ describe("IModelTransformer", () => {
     }
   }
 
+  it("process() with preserveElementIdsForFiltering set to true should re-add deleted element with same id", async () => {
+    const sourceDbPath = IModelTransformerTestUtils.prepareOutputFile(
+      "IModelTransformer",
+      "PreserveIdOnTestModel-Source.bim"
+    );
+    const sourceDb = SnapshotDb.createEmpty(sourceDbPath, {
+      rootSubject: { name: "iModelA" },
+    });
+    Subject.insert(sourceDb, IModel.rootSubjectId, "Subject1");
+    Subject.insert(sourceDb, IModel.rootSubjectId, "Subject2");
+    sourceDb.saveChanges();
+
+    const targetDbPath = IModelTransformerTestUtils.prepareOutputFile(
+      "IModelTransformer",
+      "PreserveIdOnTestModel-Target.bim"
+    );
+    const targetDb = SnapshotDb.createEmpty(targetDbPath, {
+      rootSubject: sourceDb.rootSubject,
+    });
+
+    // Execute process() so that elements from source are copied to target
+    const transformer = new IModelTransformer(sourceDb, targetDb, {
+      preserveElementIdsForFiltering: true,
+    });
+    await transformer.process();
+    targetDb.saveChanges();
+
+    let sourceContent = await getAllElementsInvariants(sourceDb);
+    let targetContent = await getAllElementsInvariants(targetDb);
+    expect(targetContent).to.deep.equal(sourceContent);
+
+    // Delete subject 1 from target
+    const code1 = Subject.createCode(
+      targetDb,
+      IModel.rootSubjectId,
+      "Subject1"
+    );
+    const targetSubjectId1 = targetDb.elements.queryElementIdByCode(code1);
+    expect(targetSubjectId1).to.not.be.undefined;
+    targetDb.elements.deleteElement(targetSubjectId1!);
+    targetDb.saveChanges();
+
+    // update subject 2 in source
+    const code2 = Subject.createCode(
+      sourceDb,
+      IModel.rootSubjectId,
+      "Subject2"
+    );
+    const targetSubjectId2 = sourceDb.elements.queryElementIdByCode(code2);
+    const subject2 = sourceDb.elements.getElement<Subject>(targetSubjectId2!);
+    subject2.description = "Subject2 Updated Description";
+    sourceDb.elements.updateElement(subject2.toJSON());
+    sourceDb.saveChanges();
+
+    // Calling process() for second time with option to preserve elements in hopes of restoring deleted element
+    const secondTransformer = new IModelTransformer(sourceDb, targetDb, {
+      preserveElementIdsForFiltering: true,
+    });
+    await secondTransformer.process(); // should not throw error: duplicate code (65547) and should re-add deleted element
+    targetDb.saveChanges();
+
+    sourceContent = await getAllElementsInvariants(sourceDb);
+    targetContent = await getAllElementsInvariants(targetDb);
+    expect(targetContent).to.deep.equal(sourceContent);
```

Comment:
can we also assert explicitly that the deleted subject 1 from target is back and has the same ID and same content?